lib/internal/main/mksnapshot.js

```diff
@@ -119,10 +120,15 @@ function requireForUserSnapshot(id) {
     err.code = 'MODULE_NOT_FOUND';
     throw err;
   }
-  if (!supportedInUserSnapshot(normalizedId)) {
+  if (isBuildingSnapshot() && !supportedInUserSnapshot(normalizedId)) {
     if (!warnedModules.has(normalizedId)) {
-      process.emitWarning(
-        `built-in module ${id} is not yet supported in user snapshots`);
+      emitWarningSync(
```

Comment:
Is there any specific reason that this warning should be emitted synchronously?

Reply:
If it's asynchronous there is a chance that users access something in the snapshot builder that's not supported and they crash before we get to process the `process.nextTick` added by `emitWarning`, so they would not be able to see the warning and just see a crash.

Reply:
Do you want to add a comment saying that?